### PR TITLE
Blog/dynamic bitmaps

### PR DESCRIPTION
**Pull request by flabbet**

### CODE REVIEW

File: website/blog/2021-07-15-dynamic-bitmaps.mdx [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+  <source src="/videos/dynamic-bitmaps.mp4" type="video/mp4"/>
+</video>
+
+## How it works
+
+Let's start with a simple algorithm
+
+```
+1. Create a new bitmap with size 0x0 (or any other size, depending on usage)
+2. Wait for coordinates and color to apply
```
Comment (Equbuxu) [inline website/blog/2021-07-15-dynamic-bitmaps.mdx:42]: What do you mean by "to apply"

File: website/blog/2021-07-15-dynamic-bitmaps.mdx [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+3. If color is not transparent (alpha = 0):
+
+  a. If X of given coordinates is bigger than current width or Y is bigger than height, Increase size to bottom right
+
+  b. Else if new colored pixel X is smaller than 0 (relative to bitmap coordinates) or Y is smaller than 0, then increase the size to top left
```
Comment (Equbuxu) [inline website/blog/2021-07-15-dynamic-bitmaps.mdx:47]: Shouldn't there be four if's for each direction?

File: website/blog/2021-07-15-dynamic-bitmaps.mdx [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+
+As you can see, the code is not trivial, it takes a lot of steps, our implementation also does a bit more stuff, like requesting clips and
+clamping the maximum size. 
+
+## Performance
+
+Our implementation is very performant since we are using fast `BitmapContext.BlockCopy` to copy and paste pixels into a new bitmap.
+
+It's almost unnoticeable in real-time, with fast mouse movement small visual jittering is visible, but not delay whatsoever.
+It all depends on the implementation, platform, native bitmap APIs, etc.
```
Comment (Equbuxu) [inline website/blog/2021-07-15-dynamic-bitmaps.mdx:215]: What is "It all" referring to

**Review by Equbuxu — CHANGES_REQUESTED** [review]

File: website/blog/2021-07-15-dynamic-bitmaps.mdx [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+
+So what was the solution? Dynamic bitmaps!
+
+## What is a dynamic bitmap?
+
+In simple words, a dynamic bitmap is a bitmap that fits the size of its content. This is crucial for the layer system.
+Here is a visual representation of how it works 
+
+<video width="650" controls>
+  <source src="/videos/dynamic-bitmaps.mp4" type="video/mp4"/>
```
Comment (CPKreuz) [inline website/blog/2021-07-15-dynamic-bitmaps.mdx:33]: Use a more up-to-date video

**Review by CPKreuz — CHANGES_REQUESTED** [review]

File: website/blog/2021-07-15-dynamic-bitmaps.mdx (hunk 2 above)
Comment (flabbet) [reply website/blog/2021-07-15-dynamic-bitmaps.mdx:47]: Nope

**Review by CPKreuz — APPROVED** [review]

**Pull request merged**